azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClient.java

```diff
@@ -92,8 +92,10 @@ private static String getResponseStringFromConn(HttpURLConnection conn) throws I
 
         if (valuesNode != null) {
             valuesNode.forEach(valueNode -> {
-                if (valueNode != null && valueNode.get("objectType").asText().equals("Group")) {
-                    final String objectID = valueNode.get("objectId").asText();
+                if (valueNode != null && valueNode.get(aadAuthenticationProperties.getUserGroup().getKey()).asText()
+                        .equals(aadAuthenticationProperties.getUserGroup().getValue())) {
```

Comment:
suggestion to extract single method for complicated condition, as well as streaming is also recommended here. Like this
`valuesNode.stream().filter(n -> n != null).filter(isGroupEquals).forEach(d -> lUserGroups .add(xxxx))`

Reply:
Updated, the value node is not a collection only iterable. Thus the stream support call.